storage/remote/queue_manager.go

```diff
@@ -1021,6 +1022,10 @@ func (t *QueueManager) shouldReshard(desiredShards int) bool {
 		level.Warn(t.logger).Log("msg", "Skipping resharding, last successful send was beyond threshold", "lastSendTimestamp", lsts, "minSendTimestamp", minSendTimestamp)
 		return false
 	}
+	if disableTimestamp := t.reshardDisableTimestamp.Load(); time.Now().Unix() < disableTimestamp {
+		level.Warn(t.logger).Log("msg", "Skipping resharding, resharding is disabled", "disabled_for", time.Until(time.Unix(disableTimestamp, 0)))
+		return false
```

Comment:
can we make the logging more explicit that resharding is disabled due to recoverable errors (vs just not being able to send successfully)? might also be useful to track the time at which we started disabling the resharding, and include that in the log line?

thinking about whether metrics for this and the other resharding blocker would be useful :thinking:

Reply:
> might also be useful to track the time at which we started disabling the resharding, and include that in the log line?

This is less easy to do; adding this would require more logic to detect whether resharding was previously enabled when setting the disabled timestamp, and would complicate the code a bit more. If you think this is a must have, I'll add it, but if it's not required I'd like to leave it out to keep the code as simple as possible.

Reply:
@cstyan I did my best at tracking this; I'm not convinced that it's 100% accurate but I think it'll be a good enough approximation of what time resharding first got disabled.